wren/compute/materialization.go

```diff
@@ -0,0 +1,175 @@
+
+	v1alpha "github.com/kaskada-ai/kaskada/gen/proto/go/kaskada/kaskada/v1alpha"
+	"github.com/kaskada-ai/kaskada/wren/customerrors"
+	"github.com/kaskada-ai/kaskada/wren/ent"
+	"github.com/kaskada-ai/kaskada/wren/ent/materialization"
+	"github.com/rs/zerolog/log"
+)
+
+func (m *Manager) StartMaterialization(ctx context.Context, owner *ent.Owner, materializationID string, compileResp *v1alpha.CompileResponse, destination *v1alpha.Destination) error {
+	subLogger := log.Ctx(ctx).With().Str("method", "manager.StartMaterialization").Str("materialization_id", materializationID).Logger()
+
+	tables, err := m.getMaterializationTables(ctx, owner, compileResp)
+	if err != nil {
+		subLogger.Error().Err(err).Msg("issue getting materialization tables")
+	}
+
+	startRequest := &v1alpha.StartMaterializationRequest{
+		MaterializationId: materializationID,
+		Plan:              compileResp.Plan,
+		Tables:            tables,
+		Destination:       destination,
+	}
+
+	computeClient := m.computeClients.ComputeServiceClient(ctx)
+	defer computeClient.Close()
+
+	subLogger.Info().
+		Interface("tables", startRequest.Tables).
+		Interface("destination", startRequest.Destination).Msg("sending start materialization request to compute backend")
+
+	_, err = computeClient.StartMaterialization(ctx, startRequest)
+	if err != nil {
+		subLogger.Error().Err(err).Msg("issue starting materialization")
+		return customerrors.NewComputeError(reMapSparrowError(ctx, err))
+	}
+
+	return nil
+}
+
+func (m *Manager) StopMaterialization(ctx context.Context, materializationID string) error {
+	subLogger := log.Ctx(ctx).With().Str("method", "manager.StopMaterialization").Str("materialization_id", materializationID).Logger()
+
+	stopRequest := &v1alpha.StopMaterializationRequest{
+		MaterializationId: materializationID,
+	}
+
+	computeClient := m.computeClients.ComputeServiceClient(ctx)
+	defer computeClient.Close()
+
+	_, err := computeClient.StopMaterialization(ctx, stopRequest)
+	if err != nil {
+		subLogger.Error().Err(err).Msg("issue stopping materialization")
+		return customerrors.NewComputeError(reMapSparrowError(ctx, err))
+	}
+
+	return nil
+}
+
+func (m *Manager) GetMaterializationStatus(ctx context.Context, materializationID string) (*v1alpha.ProgressInformation, error) {
+	subLogger := log.Ctx(ctx).With().Str("method", "manager.GetMaterializationStatus").Str("materialization_id", materializationID).Logger()
+
+	statusRequest := &v1alpha.GetMaterializationStatusRequest{
+		MaterializationId: materializationID,
+	}
+
+	computeClient := m.computeClients.ComputeServiceClient(ctx)
+	defer computeClient.Close()
+
+	statusResponse, err := computeClient.GetMaterializationStatus(ctx, statusRequest)
+	if err != nil {
+		subLogger.Error().Err(err).Msg("issue getting materialization status")
+		return nil, customerrors.NewComputeError(reMapSparrowError(ctx, err))
+	}
+
+	return statusResponse.Progress, nil
+}
+
+func (m *Manager) ReconcileMaterialzations(ctx context.Context) error {
```

Comment:
running this should make sure that all materializations in the DB are running in the compute engine